app/controllers/users/registrations_controller.rb

```diff
@@ -30,6 +30,21 @@ def new
 
   private
 
+  def after_sign_up_path_for(user)
+    if sign_up_group
+      user.ability.authorize!(:join, sign_up_group)
+      sign_up_group.add_member! user
+      sign_up_group
+    else
+      super(user)
+    end
+  end
+
+  def sign_up_group
+    return unless group_key = Array(request.referrer.match(%r{\?group_key=(.*)}))[1]
```

Comment:
Hacky yes, but I figure this will be one of the first pages we move to angular so it won't be around too long.